SCF-Session9-Question1/src/meta/utility/TestShapes.java

```diff
@@ -0,0 +1,281 @@
+package meta.utility;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+import meta.utility.Shapes.ShapeType;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class TestShapes{
+
+	/**
+	 * 
+	 */
+	@Test
+	public void rectangle_PassTest(){
+		Shapes rectangle = ShapeFactory.createShape(ShapeType.RECTANGLE);
+		List<Integer> list = Arrays.asList(4,2);
+		if(rectangle instanceof Rectangle){
+		    ((Rectangle)rectangle).setOrigin(new Point(1, 1));
+		    ((Rectangle)rectangle).setLength(list.get(0));
+		    ((Rectangle)rectangle).setBreadth(list.get(1));
+		    ((Rectangle)rectangle).setOriginDistance();
+		}
+		assertEquals(8.0, rectangle.getArea(), 0.01);
+		assertEquals(12, rectangle.getPerimeter(), 0.01);
+		assertTrue(rectangle.isPointEnclosed(new Point(2, 1)));
+		assertFalse(rectangle.isPointEnclosed(new Point(5, 5)));
+	}
+
+	@Test
+	public void square_PassTest(){
+		Shapes square = ShapeFactory.createShape(ShapeType.SQUARE);
+		List<Integer> list = Arrays.asList(4);
+		if(square instanceof Square){
+			((Square)square).setOrigin(new Point(1, 1));
+			((Square)square).setSide(list.get(0));
+			((Square)square).setOriginDistance();
+		}
+		assertEquals(16.0, square.getArea(), 0.01);
+		assertEquals(16.0, square.getPerimeter(), 0.01);
+		assertTrue(square.isPointEnclosed(new Point(2, 1)));
+		assertTrue(square.isPointEnclosed(new Point(5, 5)));
+	}
+
+	@Test
+	public void triangle_PassTest(){
```

Comment:
Test class methods should be derived  from test cases dynamically. Change these test cases as per discussion.